.circleci/config.yml

```diff
@@ -1,25 +1,132 @@
 # Javascript Node CircleCI 2.0 configuration file.
 # Check https://circleci.com/docs/2.0/language-javascript/ for more details
 version: 2
+
+references:
+  workspace: &workspace
+    ~/repo
+
+  container_config: &container_config
+    docker:
+      - image: circleci/node:8.9
+    working_directory: *workspace
+
+  win64_config: &win64_config
+    docker:
+      - image: electronuserland/builder:wine
+        environment:
+          TARGET_ARCH: x64
+    working_directory: *workspace
+
+  mac_config: &mac_config
+    macos:
+      xcode: 9.2
+    working_directory: *workspace
+
 jobs:
   build:
-    docker:
-      # CircleCI maintains a library of pre-built images
-      # documented at https://circleci.com/docs/2.0/circleci-images/
-      - image: circleci/node:7.10
-    working_directory: ~/repo
+    <<: *container_config
     steps:
       - checkout
-      # Download and cache dependencies
       - restore_cache:
           keys:
           - v1-dependencies-{{ checksum "package.json" }}
-          # fallback to using the latest cache if no exact match is found
           - v1-dependencies-
       - run: sudo yarn global add yarn@1.5.1
       - run: yarn install
       - save_cache:
           paths:
             - node_modules
           key: v1-dependencies-{{ checksum "package.json" }}
+  test:
+    <<: *container_config
+    steps:
+      - checkout
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+          - v1-dependencies-{{ checksum "package.json" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-dependencies-
+      - run: sudo yarn global add yarn@1.5.1
```

Comment:
yarn is currently at version 1.7.0
Even though there's no fundamental difference/bugfixes for us that we need, it might still be good to keep up with the versions?